src/controllers/AuthController.js

```diff
@@ -0,0 +1,94 @@
+ */
+class Auth {
+  /**
+   * Sends a user reset password link
+   *
+   * @param {Request} req - The request from the endpoint.
+   * @param {Response} res - The response returned by the method.
+   * @returns {JSON} A JSON response with a successfully message.
+   * @memberof Auth
+   */
+  static async sendResetPasswordEmail(req, res) {
+    try {
+      const { email } = req.body;
+      const user = await User.findOne({ where: { email } });
+      if (!user) {
+        return res.status(404).json(new ApiResponse(false, 404, 'User account does not exist'));
+      }
+      const { firstName, id } = user;
+      const token = generateToken({ firstName, id, email }, '24h');
+      const url = `${req.protocol}://${req.get('host')}/api/auth/reset-password?token=${token}`;
+      const response = await sendResetMail({
+        email, firstName, resetPasswordLink: url
+      });
+      if (response === true) {
+        res.status(200).json(new ApiResponse(true, 200, 'Password reset link sent successfully'));
+      } else {
+        res.status(400).json(new ApiResponse(false, 400, `${response} - Cannot send password reset link, try again`));
+      }
+    } catch (err) {
+      res.status(500).json(new ApiResponse(false, 500, `${err.message} - Cannot send password reset link, try again`));
+    }
+  }
+
+  /**
+   * Gets user new password object from the request and saves it in the database
+   *
+   * @param {Request} req - The request from the endpoint.
+   * @param {Response} res - The response returned by the method.
+   * @returns {JSON} A JSON response with the registered user and a JWT.
+   * @memberof Auth
+   */
+  static async resetPassword(req, res) {
+    try {
+      const { password } = req.body;
+      const { email } = req.params;
+      const hashPassword = hashSync(password, genSaltSync(10));
+      const [updatePassword] = await User.update({ password: hashPassword },
+        { where: { email }, returning: true });
+      if (updatePassword === 1) {
+        res.status(200).json(new ApiResponse(true, 200, 'Password has been changed successfully'));
+      } else {
+        res.status(404).json(new ApiResponse(false, 404, 'User account does not exist'));
+      }
+    } catch (err) {
+      res.status(500).json(new ApiResponse(false, 500, err.message));
+    }
+  }
+
+  /**
+   * Verify password reset link token
+   *
+   * @param {Request} req - The request from the endpoint.
+   * @param {Response} res - The response returned by the method.
+   * @returns {JSON} A JSON response with password reset link.
+   * @memberof Auth
+   */
+  static verifyPasswordResetLink(req, res) {
+    try {
+      const { token } = req.query;
+      const { email } = verifyToken(token);
+      const url = `${req.protocol}://${req.get('host')}/api/auth/password/reset/${email}`;
+      res.status(200).json(new ApiResponse(true, 200, `Goto ${url} using POST Method`));
+    } catch (err) {
+      const status = err.status || 400;
```

Comment:
If the status did not come with the error object, it's definitely a 500
```suggestion
      const status = err.status || 500;
```